internal/devices/drivers.go

```diff
@@ -38,3 +38,42 @@ func InstallWindowsDriver(ctx context.Context, vm *uvm.UtilityVM, driver string)
 	}
 	return closer, execPnPInstallDriver(ctx, vm, uvmPath)
 }
+
+// InstallKernelDriver mounts a specified kernel driver, then installs it in the UVM.
+//
+// `driver` is a directory path on the host that contains driver files for standard installation.
+// For windows this means files for pnp installation (.inf, .cat, .sys, .cert files).
+// For linux this means a vhd file that contains the drivers under /lib/modules/`uname -r` for use
+// with depmod and modprobe.
+//
+// Returns a ResourceCloser for the added mount. On failure, the mounted share will be released,
+// the returned ResourceCloser will be nil, and an error will be returned.
+func InstallKernelDriver(ctx context.Context, vm *uvm.UtilityVM, driver string) (closer resources.ResourceCloser, err error) {
+	defer func() {
+		if err != nil && closer != nil {
+			// best effort clean up allocated resource on failure
+			if releaseErr := closer.Release(ctx); releaseErr != nil {
+				log.G(ctx).WithError(releaseErr).Error("failed to release container resource")
+			}
+			closer = nil
+		}
+	}()
+	if vm.OS() == "windows" {
+		options := vm.DefaultVSMBOptions(true)
```

Comment:
Can the defer be moved and this just call InstallWindowsDriver or can both routines be refactored to call a common function?

Reply:
I just removed `InstallWindowsDriver` since it's not needed anymore with this change.